odoo/osv/expression.py

```diff
@@ -1265,14 +1265,16 @@ def __leaf_to_sql(self, eleaf):
             query = '(%s %s %s)' % (unaccent(column + cast), sql_operator, unaccent(format))
 
             if need_wildcard:
-                native_str = pycompat.to_text(right)
-                if not native_str:
-                    query = '(%s OR %s."%s" IS NULL)' % (query, table_alias, left)
-                params = ['%%%s%%' % native_str]
+                params = ['%%%s%%' % pycompat.to_text(right)]
             else:
                 field = model._fields[left]
                 params = [field.convert_to_column(right, model, validate=False)]
 
+            # what should the result be if left is null?
+            null_as = bool(right) == (operator in NEGATIVE_TERM_OPERATORS)
+            query = 'coalesce(%s, %%s)' % query
```

Comment:
I'm just guessing, so I might be saying something silly, but wouldn't it be more performant and maintainable something like this pseudocode?:

```py
if operator in NEGATIVE_TERM_OPERATORS and right is not False:
    query = "(%s OR %s IS NULL)" % (query, left)
```

... this way you'd avoid the need to provide a proper coalesced value depending on the column type (i.e. an int should be coalesced to 0, a str to "", a float to 0.0, a bool to False, etc.).

Also I imagine it's faster for postgres the `IS NOT NULL` than the `COALESCE(*,*) == whatever`.

Reply:
Not at all silly, this was and is very much an option. The original was an attempt to make the code more maintainable by having less branches / conditions (and even removing existing ones).

Reply:
@xmo-odoo Did you run any comparison between `COALESCE` and `OR IS NULL` explain plans, to get some supporting evidence?

Reply:
@odony didn't reach that point, the coalesce version broke some bits so I went with the conditional IS NULL which seem to work.